### PR TITLE
add FusedBiasLeakyRelu npu adapter

### PR DESCRIPTION
**Pull request by jayggh**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/ops/csrc/pytorch/npu/fused_bias_leakyrelu_npu.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Tensor fused_bias_leakyrelu_op_impl(Tensor& input, Tensor& bias,
+                                    Tensor& refer, int act,
+                                    int grad, float alpha, float scale);
+
+Tensor fused_bias_leakyrelu_npu(Tensor& input, Tensor& bias,
+                                Tensor& refer, int act,
+                                int grad, float alpha, float scale){
+  at::Tensor py = at::empty_like(input);
+  //forward
+  if (grad == 1){
```
Comment (grimoire) [inline mmcv/ops/csrc/pytorch/npu/fused_bias_leakyrelu_npu.cpp:15]: According to
https://github.com/open-mmlab/mmcv/blob/c69d5c4ce19efd5d597eeb6f5654dc170a74943e/mmcv/ops/fused_bias_leakyrelu.py#L174

```python
// forward
grad == 0

// backward
grad == 1
```

Comment (grimoire) [reply mmcv/ops/csrc/pytorch/npu/fused_bias_leakyrelu_npu.cpp:15]: And `act` is not used here.

**Review by grimoire — COMMENTED** [review]
Please add unit test in:
https://github.com/open-mmlab/mmcv/blob/master/tests/test_ops/test_fused_bias_leakyrelu.py

File: mmcv/ops/csrc/pytorch/npu/fused_bias_leakyrelu_npu.cpp (hunk 1 above)
Comment (jayggh) [reply mmcv/ops/csrc/pytorch/npu/fused_bias_leakyrelu_npu.cpp:15]: > And `act` is not used here.
According to [mmcv/mmcv/ops/fused_bias_leakyrelu.py], act is always 3.

File: tests/test_ops/test_fused_bias_leakyrelu.py [hunk 2]
```diff
@@ -14,36 +15,77 @@ class TestFusedBiasLeakyReLU:
...
-        cls.bias = torch.zeros(2, requires_grad=True).cuda()
+        if IS_CUDA_AVAILABLE:
+            cls.input_tensor = torch.randn((2, 2, 2, 2), requires_grad=True).cuda()
+            cls.bias = torch.zeros(2, requires_grad=True).cuda()
+        else:
+            cls.input_tensor = torch.randn((2, 2, 2, 2), requires_grad=True).npu()
+            cls.bias = torch.zeros(2, requires_grad=True).npu()
 
-    @pytest.mark.skipif(not torch.cuda.is_available(), reason='requires cuda')
+    @pytest.mark.parametrize('device', [
```
Comment (grimoire) [inline tests/test_ops/test_fused_bias_leakyrelu.py:27]: There is no parameter `device` in `test_gradient`.

**Review by grimoire — APPROVED** [review]
LGTM

File: tests/test_ops/test_fused_bias_leakyrelu.py [hunk 3]
```diff
@@ -2,48 +2,93 @@
...
+        from mmcv.ops import FusedBiasLeakyReLU
+        if IS_CUDA_AVAILABLE:
             gradcheck(
                 FusedBiasLeakyReLU(2).cuda(),
                 self.input_tensor,
-                delta=1e-4,
-                pt_atol=1e-3)
+                eps=1e-4,
+                atol=1e-3)
         else:
```
Comment (ckirchhoff2021) [inline tests/test_ops/test_fused_bias_leakyrelu.py:89]: suggest to change 'else' to 'IS_NPU_AVAILABLE'

File: tests/test_ops/test_fused_bias_leakyrelu.py [hunk 4]
```diff
@@ -2,48 +2,93 @@
...
+                    FusedBiasLeakyReLU(2).cuda(),
+                    self.input_tensor,
+                    delta=1e-4,
+                    pt_atol=1e-3)
+            elif IS_NPU_AVAILABLE:
+                gradcheck(
+                    FusedBiasLeakyReLU(2).npu(),
+                    self.input_tensor,
+                    delta=1e-4,
+                    pt_atol=1e-3)
```
Comment (zhouzaida) [inline tests/test_ops/test_fused_bias_leakyrelu.py:55]: Does NPU supports parrots?

File: tests/test_ops/test_fused_bias_leakyrelu.py [hunk 5]
```diff
@@ -2,48 +2,93 @@
...
+                    FusedBiasLeakyReLU(2).cuda(),
+                    self.input_tensor,
+                    eps=1e-4,
+                    atol=1e-3)
+            elif IS_NPU_AVAILABLE:
+                gradcheck(
+                    FusedBiasLeakyReLU(2).npu(),
+                    self.input_tensor,
+                    eps=1e-4,
+                    atol=1e-3)
```
Comment (zhouzaida) [inline tests/test_ops/test_fused_bias_leakyrelu.py:68]:
```suggestion
            gradcheck(
                FusedBiasLeakyReLU(2).to(device),
                self.input_tensor,
                eps=1e-4,
                atol=1e-3)
```

File: tests/test_ops/test_fused_bias_leakyrelu.py [hunk 6]
```diff
@@ -2,48 +2,93 @@
...
                 self.input_tensor,
-                delta=1e-4,
-                pt_atol=1e-3)
-        else:
+                eps=1e-4,
+                atol=1e-3)
+        elif IS_NPU_AVAILABLE:
             gradcheck(
-                FusedBiasLeakyReLU(2).cuda(),
+                FusedBiasLeakyReLU(2).npu(),
```
Comment (zhouzaida) [inline tests/test_ops/test_fused_bias_leakyrelu.py:91]: Same as the above comment that merges them to one case using `.to(device)` method.

File: tests/test_ops/test_fused_bias_leakyrelu.py [hunk 7]
```diff
@@ -2,48 +2,73 @@
...
+            'npu',
+            marks=pytest.mark.skipif(
+                not IS_NPU_AVAILABLE, reason='requires NPU support'))
+    ])
+    def test_gradgradient(self, device):
 
         from mmcv.ops import FusedBiasLeakyReLU
-        gradgradcheck(
-            FusedBiasLeakyReLU(2).cuda(),
+        gradcheck(
```
Comment (zhouzaida) [inline tests/test_ops/test_fused_bias_leakyrelu.py:70]: Why do we need to replace `gradgradcheck` with `gradcheck`?

**Review by zhouzaida — APPROVED** [review]

**Comment by OpenMMLab-Team** [comment]
Hi @jayggh ！First of all, we want to express our gratitude for your significant PR in the MMCV project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution！❤

**Pull request merged**